rootio/scanner.go

```diff
@@ -54,6 +54,11 @@ func (s *baseScanner) SeekEntry(i int64) error {
 	if s.err != nil {
 		return s.err
 	}
+	if s.chain {
+		if s.cur >= s.n {
+			s.nextTree(0)
```

Comment:
this works in the bench because we do `SeekEntry(0)`, which by construction is indeed the tree `0`.

but that wouldn't work if we had 2 trees (like we do) and issued `SeekEntry(12)` (which should land inside `chain.2.root`.)

so, this:
```go
s.nextTree(0)
```
should be replaced with:
```go
itree := s.findTree(i) // find the tree number in the chain that contains entry i
s.loadTree(itree)
```